### PR TITLE
rose.host_select: improve check of local host

### LINKED ISSUE #1619: rose host-select: detect local host
`rose host-select` should not ssh to a given host if it is the localhost where the command is being run.

For example, while logged on to a machine `foo`, `rose host-select foo` should not ssh.

### PR DESCRIPTION
**Pull request by matthewrmshin**
Now check for "localhost", its hostname, its fqdn hostname, and the
associated IP addresses.

Close #1619. @benfitzpatrick please test and review.

### CODE REVIEW

File: lib/python/rose/host_select.py [hunk 1]
```diff
@@ -132,6 +133,38 @@ def __init__(self, event_handler=None, popen=None):
             popen = RosePopener(event_handler=event_handler)
         self.popen = popen
         self.scorers = {}
+        self.local_host_strs = None
+
+    def get_local_host_strs(self):
+        """Return a list of names associated with the current host."""
+        if self.local_host_strs is None:
+            self.local_host_strs = []
+            for item in gethostname, getfqdn, "localhost":
```
Comment (benfitzpatrick) [inline lib/python/rose/host_select.py]: I didn't know we could do this with a `for` statement!

I think it's better to build up a list of names, then go into the `name not in...` code. That avoids having the `callable` test (and the `if name` one later).

File: lib/python/rose/host_select.py [hunk 2]
```diff
@@ -132,6 +133,38 @@ def __init__(self, event_handler=None, popen=None):
...
+                        name = item()
+                    except IOError:
+                        name = None
+                else:
+                    name = item
+                if name and name not in self.local_host_strs:
+                    self.local_host_strs.append(name)
+                    try:
+                        address = gethostbyname(name)
+                    except IOError:
```
Comment (benfitzpatrick) [inline lib/python/rose/host_select.py]: Better to use `socket.error`?

**Comment by benfitzpatrick** [comment]
We could have a test for #619 if we created a mock `ssh`. Or is it almost a case for unit tests of the host-select code?

**Comment by matthewrmshin** [comment]
All comments addressed.

**Pull request merged**